clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -218,15 +256,14 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
                 allRevokedPartitions.addAll(ownedPartitions.subList(minQuota, ownedPartitions.size()));
                 // this consumer is potential maxQuota candidate since we're still under the number of expected members
                 // with more than the minQuota partitions. Note, if the number of expected members with more than
-                // the minQuota partitions is 0, it means minQuota == maxQuota, so they won't be put into unfilledMembers
+                // the minQuota partitions is 0, it means minQuota == maxQuota, and there are no potentially unfilled
                 if (numMembersAssignedOverMinQuota < expectedNumMembersAssignedOverMinQuota) {
-                    unfilledMembers.add(consumer);
+                    potentiallyUnfilledMembersAtMinQuota.add(consumer);
```

Comment:
This is part of fix #3 -- basically we have to handle these members separately, once they get up to `minQuota` they are only "potentially" unfilled. Once the last member allowed reaches `maxQuota`, all of these `minQuota` members are suddenly considered filled. cc @showuon

Reply:
Honestly it took me quite a while to understand the fix :P After understanding that I think maybe it's better to rename these two collections more explicitly:

1) `unfilledMembers` -> `MembersWithLessThanMinQuotaPartitions`.
2) `potentiallyUnfilledMembersAtMinQuota` -> `MembersWithExactMinQuotaPartitions`.

And also (since the maxQuota is always either == minQuota or minQuota + 1):

3) `expectedNumMembersAssignedOverMinQuota` -> `expectedNumMembersWithMaxQuota` 
4) `numMembersAssignedOverMinQuota` -> `numMembersWithMaxQuota`

Reply:
Ack on the first two renamings, though I'd still want to prefix them with `unfilled` to emphasize that these structures only hold members that may potentially be assigned one or more partitions. ie, if `minQuota == maxQuota`, then `potentiallyUnfilledMembersAtMinQuota` should actually be empty, in which case `MembersWithExactMinQuotaPartitions` doesn't quite make sense. I'll clarify this in the comments as well.

Reply:
For the 3 & 4th suggested renamings, it's a bit subtle but this would actually be incorrect. In the case `minQuota == maxQuota`, the `expectedNumMembersAssignedOverMinQuota` variable will evaluate to 0, which would not make sense if it was called `expectedNumMembersWithMaxQuota`. Of course we could go through a tweak the logic for this case, but I'd prefer not to mix that into this PR. For now I'll just clarify in the comments for these variables

Reply:
Ack.